### PR TITLE
Add agent to agent feature

### PR DESCRIPTION
**Pull request by johntdyer**

### CODE REVIEW

File: agent_agent.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	var target map[string][]AgentAgent
+	if dErr := c.decodeJSON(resp, &target); dErr != nil {
+		return nil, fmt.Errorf("Could not decode JSON response: %v", dErr)
+	}
+	return &target["test"][0], nil
+}
+
+// CreateAgentAgent - Create an agent to agent test
+func (c Client) CreateAgentAgent(t AgentAgent) (*AgentAgent, error) {
+	resp, err := c.post("/tests/agent-to-server/new", t, nil)
```
Comment (william20111) [inline agent_agent.go:66]: `agent-to-agent`

File: agent_agent.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	var target map[string][]AgentAgent
+	if dErr := c.decodeJSON(resp, &target); dErr != nil {
+		return nil, fmt.Errorf("Could not decode JSON response: %v", dErr)
+	}
+	return &target["test"][0], nil
+}
+
+//DeleteAgentAgent - delete agent to agent test
+func (c *Client) DeleteAgentAgent(id int) error {
+	resp, err := c.post(fmt.Sprintf("/tests/agent-to-server/%d/delete", id), nil, nil)
```
Comment (william20111) [inline agent_agent.go:82]: `agent-to-agent`

File: agent_agent.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+// CreateAgentAgent - Create an agent to agent test
+func (c Client) CreateAgentAgent(t AgentAgent) (*AgentAgent, error) {
+	resp, err := c.post("/tests/agent-to-server/new", t, nil)
+	if err != nil {
+		return &t, err
+	}
+	if resp.StatusCode != 201 {
+		return &t, fmt.Errorf("failed to create agent server, response code %d", resp.StatusCode)
```
Comment (william20111) [inline agent_agent.go:71]: log message tweak

File: agent_agent.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+//DeleteAgentAgent - delete agent to agent test
+func (c *Client) DeleteAgentAgent(id int) error {
+	resp, err := c.post(fmt.Sprintf("/tests/agent-to-server/%d/delete", id), nil, nil)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != 204 {
+		return fmt.Errorf("failed to delete agent server, response code %d", resp.StatusCode)
```
Comment (william20111) [inline agent_agent.go:87]: log message tweak

File: agent_agent.go [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+	}
+	return nil
+}
+
+// UpdateAgentAgent - update agent to agent test
+func (c *Client) UpdateAgentAgent(id int, t AgentAgent) (*AgentAgent, error) {
+	resp, err := c.post(fmt.Sprintf("/tests/agent-to-server/%d/update", id), t, nil)
```
Comment (william20111) [inline agent_agent.go:94]: `agent-to-agent`

File: agent_agent.go [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+	if err != nil {
+		return &t, err
+	}
+	if resp.StatusCode != 200 {
+		return &t, fmt.Errorf("failed to update agent server, response code %d", resp.StatusCode)
```
Comment (william20111) [inline agent_agent.go:99]: log tweak required

File: agent_agent.go (hunk 1 above)
Comment (johntdyer) [reply agent_agent.go]: ack

File: agent_agent.go (hunk 4 above)
Comment (johntdyer) [reply agent_agent.go]: ty

**Comment by johntdyer** [comment]
updated PR, thanks for the review !

**Comment by johntdyer** [comment]
_Tests... turns out they should pass... who knew_ ;-P

**Pull request merged**